java/ingest/ingest-batcher-core/src/main/java/sleeper/ingest/batcher/IngestBatcher.java

```diff
@@ -76,27 +76,58 @@ public static Builder builder() {
 
     public void batchFiles() {
         Instant time = timeSupplier.get();
-        store.getPendingFilesOldestFirst().stream()
-                .collect(Collectors.groupingBy(FileIngestRequest::getTableName, LinkedHashMap::new, toList()))
-                .forEach((tableName, inputFiles) -> batchTableFiles(tableName, inputFiles, time));
+        if (store.getPendingFilesOldestFirst().isEmpty()) {
+            LOGGER.info("No pending files found");
+        } else {
+            store.getPendingFilesOldestFirst().stream()
```

Comment:
Can we do `store.getPendingFilesOldestFirst()` before the if statement so that we're not calling the method twice in the case that it's not empty?